substratevm/src/com.oracle.svm.core.posix/src/com/oracle/svm/core/posix/JavaNetNetworkInterface.java

```diff
@@ -184,6 +186,39 @@ public void toLog(Log log) {
         }
     }
 
+    static boolean isUp0(String name, int ind) throws SocketException {
+        int flags = getFlags0(name);
+        return CTypeConversion.toBoolean(flags & NetIf.IFF_UP())
+                && CTypeConversion.toBoolean(flags & NetIf.IFF_RUNNING());
+    }
+
+    static int getFlags0(String ifname) throws SocketException {
```

Comment:
Similarly, this looks like a translation of http://hg.openjdk.java.net/jdk8u/jdk8u/jdk/file/15a6dced59c3/src/solaris/native/java/net/NetworkInterface.c#l620.

The point of inter-lining the original C code is that the reader does not want to have to understand what the code is doing, or have to understand how you have restructured the code.  It is hard enough to check that the Java code is a faithful translation of the C code.